### PR TITLE
BUG: Fix MultiIndex .loc with all numpy arrays

### LINKED ISSUE #19686: BUG: MultiIndex .loc fails with all numpy arrays
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np

a = [10,20,30]
b = [1,2,3]
index = pd.MultiIndex.from_product([a,b])
df = pd.DataFrame(np.arange(len(index)), index=index, columns=['Data'])

slice1 = df.loc[([10,20],            [2,3]),           'Data']
slice2 = df.loc[(np.array([10,20]),  [2,3]),           'Data']
slice3 = df.loc[([10,20],            np.array([2,3])), 'Data']
slice4 = df.loc[(np.array([10,20]),  np.array([2,3])), 'Data']

```
#### Problem description

slice4 in the sample fails with TypeError: '(array([10, 20]), array([2, 3]))' is an invalid key.
slice1-3 all work as expected.

#### Expected Output

expect slicing to work with numpy arrays the same as lists.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 45 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.22.0
pytest: 2.9.2
pip: 8.1.2
setuptools: 27.2.0
Cython: 0.24.1
numpy: 1.14.0
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 5.1.0
sphinx: 1.4.6
patsy: 0.4.1
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: None
matplotlib: 2.1.2
openpyxl: 2.3.2
xlrd: None
xlwt: 1.1.2
xlsxwriter: 0.9.3
lxml: 3.6.4
bs4: 4.5.1
html5lib: 0.9999999
sqlalchemy: 1.0.13
pymysql: None
psycopg2: None
jinja2: 2.8
s3fs: 0.1.2
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by spacesphere**
- [x] closes #19686 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -767,6 +767,7 @@ Indexing
 - Bug in :class:`IntervalIndex` where empty and purely NA data was constructed inconsistently depending on the construction method (:issue:`18421`)
 - Bug in :func:`IntervalIndex.symmetric_difference` where the symmetric difference with a non-``IntervalIndex`` did not raise (:issue:`18475`)
 - Bug in :class:`IntervalIndex` where set operations that returned an empty ``IntervalIndex`` had the wrong dtype (:issue:`19101`)
+- Bug in indexing where nested indexers having only numpy arrays are handled incorrectly (:issue:`19686`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: can you move to the mult-index section

File: pandas/tests/indexing/test_loc.py [hunk 2]
```diff
@@ -711,3 +711,36 @@ def test_identity_slice_returns_new_object(self):
 
         original_series[:3] = [7, 8, 9]
         assert all(sliced_series[:3] == [7, 8, 9])
+
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: can you add Index here as well.

File: pandas/tests/indexing/test_loc.py [hunk 3]
```diff
@@ -711,3 +711,36 @@ def test_identity_slice_returns_new_object(self):
...
+    @pytest.mark.parametrize('indexer_type_1',
+                             (list, tuple, set, slice, np.ndarray, Series))
+    @pytest.mark.parametrize('indexer_type_2',
+                             (list, tuple, set, slice, np.ndarray, Series))
+    def test_loc_getitem_nested_indexer(self, indexer_type_1, indexer_type_2):
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: add the gh issue number & a 1-liner of what you are testing

File: pandas/tests/indexing/test_loc.py [hunk 4]
```diff
@@ -711,3 +711,36 @@ def test_identity_slice_returns_new_object(self):
...
+            return indexer_type(keys)
+
+        a = [10, 20, 30]
+        b = [1, 2, 3]
+        index = pd.MultiIndex.from_product([a, b])
+        df = pd.DataFrame(np.arange(len(index)), index=index, columns=['Data'])
+
+        keys = ([10, 20], [2, 3])
+        types = (indexer_type_1, indexer_type_2)
+        # check indexers with all the combinations of nested objects
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: blank line here

File: pandas/tests/indexing/test_loc.py [hunk 5]
```diff
@@ -711,3 +711,36 @@ def test_identity_slice_returns_new_object(self):
...
+                return slice(*keys)
+            return indexer_type(keys)
+
+        a = [10, 20, 30]
+        b = [1, 2, 3]
+        index = pd.MultiIndex.from_product([a, b])
+        df = pd.DataFrame(np.arange(len(index)), index=index, columns=['Data'])
+
+        keys = ([10, 20], [2, 3])
+        types = (indexer_type_1, indexer_type_2)
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: I *think* you might be able to use ``np.ix_`` for this, we also have some utilities in pandas/core/indexing.py

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_loc.py [hunk 6]
```diff
@@ -711,3 +711,42 @@ def test_identity_slice_returns_new_object(self):
...
+            if indexer_type == np.ndarray:
+                return np.array(keys)
+            if indexer_type == slice:
+                return slice(*keys)
+            return indexer_type(keys)
+
+        a = [10, 20, 30]
+        b = [1, 2, 3]
+        index = pd.MultiIndex.from_product([a, b])
+        df = pd.DataFrame(np.arange(len(index)), index=index, columns=['Data'])
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: you need to add ``dtype='int64'`` otherwise this won't match on windows (as np.arange defaults to int32)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_loc.py (hunk 6 above)
Comment (PoppyBagel) [reply pandas/tests/indexing/test_loc.py]: good catch, thanks

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green.

**Comment by PoppyBagel** [comment]
@jreback all the checks have already finished

**Comment by jreback** [comment]
thanks!

**Pull request merged**